dev_docs/key_concepts/feature_privileges.mdx

```diff
@@ -0,0 +1,435 @@
+
+```
+{
+    ...,
+    privileges: {
+        all: {
+            ...,
+            ui: ['view', 'create', 'edit', 'delete', 'assign'],
+        },
+        read: {
+            ...,
+            ui: ['view'],
+        }
+    }
+}
+```
+
+Once done, you can now access this within the public part kibana plugin code using the Kibana Context Provider as follows:
+
+```tsx
+export const MyPluginComponent: React.FC = () => {
+  const kibana = useKibana<CoreStart>();
+
+  return (
+    <EuiPageTemplate>
+      <EuiPageTemplate.Section grow={false} color="subdued" bottomBorder="extended">
+        <EuiTitle size="l">
+          <h1>Feature Privileges Example</h1>
+        </EuiTitle>
+      </EuiPageTemplate.Section>
+      <EuiPageTemplate.Section grow={false} color="subdued" bottomBorder="extended">
+        <EuiText>
+          <p>Your privileges</p>
+        </EuiText>
+        <EuiSpacer />
+        {Object.entries(
+          kibana.services.application!.capabilities[FEATURE_PRIVILEGES_PLUGIN_ID]
+        ).map(([capability, value]) => {
+          return value === true ? (
+            <div key={capability}>
+              <EuiHealth color="success">{capability}</EuiHealth>
+              <EuiSpacer />
+            </div>
+          ) : null;
+        })}
+      </EuiPageTemplate.Section>
+    </EuiPageTemplate>
+  );
+};
+```
+
+### Control access to API endpoints and other server side features
+
+Similarly for API access, lets modify the server side of our plugin by adding a couple of API routes as follows:
+
+```ts
+public setup(core: CoreSetup, deps: FeatureControlExampleDeps) {
+  // ...
+  const router = core.http.createRouter();
+    router.get(
+      {
+        path: '/internal/my_plugin/read',
+        validate: false,
+      },
+      async (context, request, response) => {
+        return response.ok({
+          body: {
+            time: new Date().toISOString(),
+          },
+        });
+      }
+    );
+    router.get(
+      {
+        path: '/internal/my_plugin/sensitive_action',
+        validate: false,
+      },
+      async (context, request, response) => {
+        return response.ok({
+          body: {
+            time: new SecretClass().secretResponse(),
+          },
+        });
+      }
+    );
+}
+```
+
+Here we've added two routes, one which returns the current date-time in ISO 8601 format and another which returns a sensitive response that we'd like to protect.
+
+To do so, we need to modify the configuration object for our route to look like this:
+
+```
+router.get(
+      {
+        path: '/internal/my_plugin/sensitive_action',
+        validate: false,
+        options: {
+          tags: ['access:my_closed_example_api'],
+        },
+      },
+      async (context, request, response) => {
+        return response.ok({
+          body: {
+            time: new Date().toISOString(),
```

Comment:
I think this is the function we didn't want to secure, should it be `secretResponse()`?